### PR TITLE
Support for Volumes

### PR DESCRIPTION
**Pull request by ddoowa**
Added Functionality to define volumes to attach the function pods to, also refactored the process to define secrets to re-use the same underlying function.

example volume definitions:
```yaml
functions:
  my-function:
    handler: handler.myFunction
    volumes:
      - name: my-volume
        persistentVolumeClaim:
            claimName: my-volume
    volumeMounts:
      - mountPath: /foo/bar
         name: my-volume
```

### CODE REVIEW

File: lib/deploy.js [hunk 1]
```diff
@@ -58,6 +58,21 @@ function parseEnv(src) {
   }
   return res;
 }
+function createVolumes(container, funcs, volumes) {
```
Comment (andresmgot) [inline lib/deploy.js]: this function is a bit confusing, it's appending the volumes to the function and the container but it seems that the paremeter are unrelated (and not sure why it's called created?)

If you want to do something like this, my recomendation is to return an object here instead `function parseVolumes(volumes) => {volumes, volumeMounts}` so the caller can use the result and attach them to the proper containers or deployments

File: .eslintrc.js [hunk 2]
```diff
@@ -7,7 +7,8 @@ module.exports = {
     "prefer-rest-params": "off",
     "react/require-extension" : "off",
     "import/no-extraneous-dependencies" : "off",
-    "no-console": "off"
+    "no-console": "off",
+    "no-param-reassign": "off",
```
Comment (andresmgot) [inline .eslintrc.js]: please, don't disable linter rules for the whole project. If there is a case in which you need to disable this rule, do it inline (but I think it's not necessary)

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless-kubeless",
-  "version": "0.8.3",
+  "version": "0.8.4",
```
Comment (andresmgot) [inline package.json]: it's better to update the minor version since we are adding a new feature (0.9.0)

File: test/kubelessDeploy.test.js [hunk 4]
```diff
@@ -387,9 +387,45 @@ describe('KubelessDeploy', () => {
         },
       }));
       return expect( // eslint-disable-line no-unused-expressions
-        kubelessDeploy.deployFunction()
+          kubelessDeploy.deployFunction()
       ).to.be.fulfilled;
     });
+    it('should deploy a function with volumes', () => {
```
Comment (andresmgot) [inline test/kubelessDeploy.test.js]: thanks for the test!

File: lib/deploy.js (hunk 1 above)
Comment (ddoowa) [reply lib/deploy.js]: changing the structure of the definition removed the need for a parse function as the attributes can be directly applied, thanks

File: package.json (hunk 3 above)
Comment (ddoowa) [reply package.json]: done

File: test/kubelessDeploy.test.js (hunk 4 above)
Comment (ddoowa) [reply test/kubelessDeploy.test.js]: Tests are awesome :)

**Comment by ddoowa** [comment]
> Thanks for the PR @ddoowa!
> 
> I have a few suggestions, apart from that I think it's better to diverge as less as possible from the format that a pod expect for volumes. In particular I would suggest:
> 
> ```
> functions:
>   my-function:
>     handler: handler.myFunction
>     volumes:
>       - name: my-volume
>         persistentVolumeClaim:
>           claimName: my-volume
>     volumeMounts:
>       - mountPath: /foo/bar
>         name: my-volume
> ```
> 
> That way it's easier for people to know the format and it's easier to use in the code (you don't need to parse and translate the information from the function spec).

That sounds even better, not sure why i didn't think of it, but i have now made the necessary changes to support this format, and addressed other comments, looking forward to the merge.

**Pull request merged**